src/main/java/seedu/address/logic/trie/CommandTrie.java

```diff
@@ -0,0 +1,131 @@
+package seedu.address.logic.trie;
+
+import java.util.Map;
+import java.util.Set;
+
+import seedu.address.logic.commands.CommandCollection;
+
+
+/**
+ * Class for the autocomplete command function
+ * The class holds the trie for command words
+ */
+public class CommandTrie implements Trie {
+
+    private Node root = null;
+    private Set<String> commandSet;
+    private Map<String, String> commandMap;
+    private CommandCollection commandCollection;
+
+    public CommandTrie () {
+        commandCollection = new CommandCollection ();
+        commandSet = commandCollection.getCommandSet();
+        commandMap = commandCollection.getCommandMap();
+
+        for (String command : commandSet) {
+            this.insert(command);
+        }
+    }
+
+    /**
+     * Indicates whether or note a node is a leaf
+     */
+    public boolean isLeaf(Node current) {
+        if (!current.hasNext() && !current.hasChild()) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @param input key to autocomplete
+     * @return input if the command is not found, otherwise String representation of command word
+     */
+    public String attemptAutoComplete (String input) throws NullPointerException {
```

Comment:
can you possibly increase the SLAP (Single level of abstraction principle) in this method?